internal/test/helpers.go

```diff
@@ -214,6 +217,29 @@ func SetupCreateUserDevice(t *testing.T, testUserID string, ddID string, metadat
 	return ud
 }
 
+func SetupCreateUserDeviceForMiddleware(t *testing.T, userDeviceID, userID string, ddID string, metadata *[]byte, vin string, pdb db.Store) models.UserDevice {
```

Comment:
I'm surprised that we can't use something off the shelf. What does this offer beyond `SetupCreateUserDevice`? It's not clear to me.

Reply:
`SetupCreateUserDevice` generates the user id using `ksuid.New()` but I wanted to be able to pass it in